lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp

```diff
@@ -557,11 +557,19 @@ int WiFiClientSecure_light::_run_until(unsigned target, bool blocking) {
     DEBUG_BSSL("_run_until: Not connected\n");
     return -1;
   }
+  lastInActivity = lastOutActivity = millis();
   for (int no_work = 0; blocking || no_work < 2;) {
     if (blocking) {
       // Only for blocking operations can we afford to yield()
       optimistic_yield(100);
     }
+    
+    unsigned long t = millis();
+    if (t-lastInActivity >= ((int32_t) this->_loopTimeout*1000UL) ){

```

Comment:
This comparison does not work when timer becomes negative after 27 days. Please use `TimeReached()` instead which does the correct comparison

Reply:
Thanks @s-hadinger , I'll look into that

Reply:
As `TimeReached()` function is support function for tasmota flow, I just implemented same comparison in this fix